### PR TITLE
41 - Backend Base Project configuration

### PR DESCRIPTION
**Pull request by Lombardoh**
Basic project configuration

main points:

-basic folder structure
-a basic user schema in order to be able to test, created using sqlmodel
-db folder will be used for connecting to DB in production and also can be used to generate a local db for testing
-versioning thru .env file (file needs to be provided or created with proper variables)
-test are working with testClient, there are 2 endpoints to test one with a mock user and the other will serve for general information (including version)
-routers folder added in order to organize the endpoints
-dockerfile
-readme completed with instructions to deploy in docker and as a local app for dev

### CODE REVIEW

**Comment by jpmckinney** [comment]
Hi, the Lint workflow is failing because it expects dependencies to be managed as documented here: https://ocp-software-handbook.readthedocs.io/en/latest/python/requirements.html

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -18,4 +18,4 @@ jobs:
       - shell: bash
         run: curl -s -S --retry 3 $BASEDIR/tests/script.sh | bash -
       - run: pip install -r requirements_dev.txt
-      - run: pytest /tmp/test_requirements.py
+      - run: pytest tests -W error
```
Comment (jpmckinney) [inline .github/workflows/lint.yml:21]: We typically run an application's tests in a `ci.yml` file: https://ocp-software-handbook.readthedocs.io/en/latest/python/ci.html#python-ci-templates

The `pytest /tmp/test_requirements.py` line is running this test: https://github.com/open-contracting/standard-maintenance-scripts/blob/main/tests/test_requirements.py

That test guarantees that requirements are being managed correctly, as documented here: https://ocp-software-handbook.readthedocs.io/en/latest/python/requirements.html

Comment (Lombardoh) [reply .github/workflows/lint.yml]: Hello James. Thanks for the documentation, let me take a look and try to fix the rest of the errors I got

**Comment by jpmckinney** [comment]
Hi @Lombardoh, you seemed to be having trouble getting the tests to pass, so I added a commit.

In short: If you `import` a library in application code, it needs to be listed in `requirements.in`

If you `import` any other library in test code, it needs to be in `requirements_dev.in`

**Comment by jpmckinney** [comment]
I see many commits relating to pre-commit issues. I *highly* recommend running:

```
pip install pre-commit
pre-commit install
```

That way, a pre-commit git hook will warn before committing.

You can also configure your editor to run `pre-commit run` whenever you save a file, for example.

**Comment by jpmckinney** [comment]
FYI: The pre-commit.ci check failed, which is fine, as it often happens due to their infrastructure ("Temporary failure in name resolution" when trying to reach PyPI). This check is configured to be ignored, so the overall PR gets a check mark.

**Pull request merged**